### ISSUE TITLE
amp-consent: Graphical controls missing an appropriate text description

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31153]
## WCAG Level
Level A

## Priority
Low

## Type
Images

## Pages/screens/components affected
* [amp-consent](https://amp.dev/documentation/components/amp-consent/)

## Description
When images or icons that are part of an interactive control do not have an appropriate text description, anyone who cannot see images  will have difficulty understanding the meaning/purpose of the control.

The `amp-consent` modal dialogs currently have an "X" close control, which simply uses the text character as its content. It is announced as "X - button" by assistive technologies.

![An example consent modal dialog, with the 'X' control highlighted](https://user-images.githubusercontent.com/54933393/99180902-4aaaa200-2722-11eb-8f75-ff993f645355.png)

## User impact
While in this case the impact on assistive technology users is low, as they will likely be able to guess that "X" represents a close control, this is still a defect that should be remediated.

## Required solution
Make sure that every image that conveys information or is used to perform an action has an appropriate text description. The text description must be clear and relevant to the context in which the image is being used.

The current markup for the "X" close control is a "faked" button using `<div>` element, containing the "X" character.

```
<div class="dismiss-button" role="button" tabindex="0" on="tap:myConsent.dismiss">X</div>
```

The simplest option in this case is to add an explicit `aria-label` to the control. This will be used as the accessible name of the control that is announced by assistive technologies, and act as the text alternative for the "X" icon. We would also recommend, unless there's a good reason against it, using an actual `<button>` element.

```
<button class="dismiss-button" aria-label="Close" on="tap:myConsent.dismiss">X</button>
```

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Check that all images are provided with appropriate alternative text.
2. Check that the alternative text acts as a suitable equivalent for the image.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website  has been tested for the same issue.
* All issues identified throughout the rest of the website  have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 1.1.1 Non-text Content (Level A)](https://www.w3.org/TR/WCAG21/#non-text-content)

## More information
* [W3C: Image concepts](https://www.w3.org/WAI/tutorials/images/decision-tree/)
* [W3C: An alt decision tree](https://www.w3.org/WAI/tutorials/images/decision-tree/)
* [W3C: HTML 5.2 Requirements for providing text to act as an alternative for images](https://www.w3.org/TR/html52/semantics-embedded-content.html#alt-text)

## Test data
Test date: November 2020
